lib/class-wp-rest-global-styles-controller-gutenberg.php

```diff
@@ -458,14 +497,219 @@ protected function get_available_actions( $post, $request ) {  // phpcs:ignore V
+			'isGlobalStylesUserThemeJSON' => true,
+		);
+
+		// Include settings and styles if provided.
+		if ( isset( $request['settings'] ) ) {
+			$content['settings'] = $request['settings'];
+		}
+		if ( isset( $request['styles'] ) ) {
+			$content['styles'] = $request['styles'];
+		}
+
+		$post_id = wp_insert_post(
+			array(
+				'post_content' => wp_json_encode( $content ),
+				'post_status'  => 'publish',
+				'post_title'   => $title,
+				'post_type'    => $this->post_type,
+				'post_name'    => 'wp-style-variation-' . sanitize_title( $title ),
+				'meta_input'   => array(
+					'_wp_style_variation' => true,
+				),
+			),
+			true
+		);
+
+		if ( is_wp_error( $post_id ) ) {
+			return $post_id;
+		}
+
+		$post     = get_post( $post_id );
+		$response = $this->prepare_item_for_response( $post, $request );
+		$response = rest_ensure_response( $response );
+		$response->set_status( 201 );
+		$response->header( 'Location', rest_url( sprintf( '%s/%s/%d', $this->namespace, $this->rest_base, $post_id ) ) );
+
+		return $response;
+	}
+
+	/**
+	 * Checks if a given request has access to delete a global style.
+	 *
+	 * @since 7.0.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has delete access, WP_Error object otherwise.
+	 */
+	public function delete_item_permissions_check( $request ) {
+		$post = $this->get_post( $request['id'] );
+		if ( is_wp_error( $post ) ) {
+			return $post;
+		}
+
+		// Only allow deleting style variations, not the main global styles.
+		$is_style_variation = get_post_meta( $post->ID, '_wp_style_variation', true );
+		if ( ! $is_style_variation ) {
+			return new WP_Error(
+				'rest_cannot_delete_global_styles',
+				__( 'Sorry, you can only delete style variations, not the main global styles.', 'gutenberg' ),
+				array( 'status' => 403 )
+			);
+		}
+
+		if ( ! current_user_can( 'edit_theme_options' ) ) {
+			return new WP_Error(
+				'rest_cannot_delete_global_styles',
+				__( 'Sorry, you are not allowed to delete this global style.', 'gutenberg' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Deletes a style variation.
+	 *
+	 * @since 7.0.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function delete_item( $request ) {
+		$post = $this->get_post( $request['id'] );
+		if ( is_wp_error( $post ) ) {
+			return $post;
+		}
+
+		$force = isset( $request['force'] ) ? (bool) $request['force'] : false;
+
+		$previous = $this->prepare_item_for_response( $post, $request );
+		$result   = wp_delete_post( $post->ID, $force );
+
+		if ( ! $result ) {
+			return new WP_Error(
+				'rest_cannot_delete',
+				__( 'The variation cannot be deleted.', 'gutenberg' ),
+				array( 'status' => 500 )
```

Comment:
A 500 doesn't make much sense here. We could use 403, 404, or even 409. Might be nice to have a more specific mapping as this error comes directly from wpdb IIRC.